### ISSUE TITLE
[AIR] Add progress bar for training 

### ISSUE THREAD

**Issue by xwjiang2010** [ray-project/ray#29314]
## User Flow
I am training a model, I want to see the current progress and how much is left of the work. If possible I would also like to see some key metrics attached to the end of the progress bar that is being updated with each round of new results. Something like:
https://devblog.pytorchlightning.ai/super-charged-progress-bars-with-rich-lightning-669653d6ab97

## Current state
AIR is showing a trial table (the same UI as if a single trial Tune) that shows "RUNNING" state and some key metrics with cut-offs every few seconds. There are also other not so relevant information like node_ip etc. The issue is that this table is not tailored to the train case.

## Proposal
Add an **option** for user to instead use progress bar. This will automatically **disable** the Tune type of logging.

**API** 
Provide an in-hourse Callback implementation:
```
pbc = ProgressBarCallback(num)  # number of times that `report` is expected to be called by the training function.
Trainer = TorchTrainer(train_loop_per_worker=..., run_config=RunConfig(callbacks=[pbc]))
```

**Comment by xwjiang2010** [comment]
cc @jiaodong @rkooo567 @amogkam @richardliaw

**Comment by rkooo567** [comment]
This SGTM. Maybe it'd be great if you can also share some e2e code example writing progress bar using the callback?

**Comment by xwjiang2010** [comment]
Reassigning to Jun as I am going on a leave.

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
